src/main/java/com/karankumar/bookproject/backend/entity/BaseEntity.java

```diff
@@ -24,13 +24,14 @@
 import javax.persistence.Id;
 import javax.persistence.MappedSuperclass;
 import java.util.Objects;
+import java.util.Optional;
 
 @MappedSuperclass
 public abstract class BaseEntity {
     @Id
     @GeneratedValue(strategy = GenerationType.SEQUENCE)
     @Getter
-    private Long id;
+    private Optional<Long> id = Optional.ofNullable(null);
```

Comment:
Using optional for class variables is not a good idea, it kind of say this member may or may not exist but actually it will exist with different values. So we can remove the optional for class variables.